SmartAppointmentSystem.Data/AppointmentContextFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Design;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.IO;
+
+namespace SmartAppointmentSystem.Data
+{
+    public class AppointmentContextFactory : IDesignTimeDbContextFactory<AppointmentContext>
+    {
+        public AppointmentContext CreateDbContext(string[] args)
+        {
+            var basePath = Directory.GetCurrentDirectory();
```

Comment:
If you move you connection string, credentials to appsettings from environment variables, you won't need DesignTime DbContextFactory. We can discuss this later, but keep this in mind for now, or take notes of it.